biochemist-python/chapters/File_Parsing.ipynb

```diff
@@ -0,0 +1,602 @@
+    "        words = PROTEIN_ATOM_line.split(':')\n",
+    "        print(words)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "From this `print` statement, we now see that we have a list called words, where we have split `PROTEIN_ATOM_line`.  The number of atoms in the protein is actually the second element of this list, so we can now save it as a new variable."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "atoms = words[1]\n",
+    "print(atoms)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The HIV protease structure in 4eyr.pdb is a dimer. Let's find the number of atoms in each monomer unit. If we now try to do a math operation on atoms, we get an error message.  Why do you think that is?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "atoms / 2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Even though `atoms` looks like a number to us, it is really a string, so we can not add an integer to it.  We need to change the data type of energy to a float. This is called *casting*."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "atoms = float(atoms)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now it will work.  If we thought ahead we could have changed the data type when we assigned the variable originally."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "atoms = float(words[1])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"exercise-this-is-a-title exercise\">\n",
+    "<p class=\"exercise-title\">Exercise on file parsing</p>\n",
+    "<p>Use the provided 7tim.pdb file, which includes structural information and data about a complex of of a small molecule bound to the enzyme, triose phosphate isomerase. Find the resolution of the structure, teh total number of protein atoms, the total number of heterogroup atoms, and the identify the small molecule (hint: HETNAM) by abbreviation and full name. Remember - you need to use os.join to be able to read the file. Your output should look like this:</p>\n",
+    "\n",
+    "<p>RESOLUTION : 1.90 ANGSTROMS  </p>  \n",
+    "<p>PROTEIN ATOMS : 3766  </p>\n",
+    "<p>HETEROGEN ATOMS : 20  </p>\n",
+    "<p>HETNAM Abbreviation: PGH Full name: PHOSPHOGLYCOLOHYDROXAMIC </p>\n",
+    "    \n",
+    "```{admonition} Solution\n",
+    ":class: dropdown\n",
+    "<p>This is one possible solution for the 7tim.pdb parsing exercise.</p>\n",
+    "    \n",
+    "```python\n",
+    "with open(protein_file,\"r\") as outfile2:\n",
```

Comment:
The technique we listed as an "alternate" technique for opening files is used here. Either it needs to use `open` and `close`, or the earlier portion of the lesson needs to use the same syntax as this solution.

Reply:
I changed the earlier portion of the lesson to include the with context manager.